Build error on minified_zutil.h on GDAL 3.7.0 on Gentoo and derivatives

<!--
IMPORTANT: Do NOT use GitHub to post any questions or support requests!
           They will be closed immediately and ignored.

Questions should go to the gdal-dev mailing list at
https://lists.osgeo.org/mailman/listinfo/gdal-dev
or other support forums. GitHub issues are for bug
reports and suggestions for new features.

If you think there is an issue with coordinate order in GDAL 3.0, then it is
likely an intended behavior. See https://github.com/OSGeo/gdal/issues/1974
for more explanations.

The GDAL project is made of contributions from various individuals and
organizations, each with their own focus. The issue you are facing is
not necessarily in the priority list of those contributors and consequently
there is no guarantee that it will be addressed in a timely manner.
If this bug report or feature request is high-priority for you, and you
cannot address it yourself, we suggest engaging a GDAL developer or support
organisation and financially sponsoring a fix.

-->

## Expected behavior and actual behavior.

compile and build gdal-3.7.0

## Steps to reproduce the problem.

use ninja to build and seeing this error:

`[41/1099] /usr/bin/x86_64-pc-linux-gnu-gcc -DDONT_DEPRECATE_SPRINTF -DGDAL_COMPILATION -DHAVE_SSE_AT_COMPILE_TIME   -mtune=generic -O2 -pipe -fvisibility=hidden -fPIC -Wall -Wextra -Winit-self -Wunused-parameter -Wmissing-prototypes -Wmissing-declarations -Wlogical-op -Wshadow -Wmissing-include-dirs -Wformat -Werror=format-security -Wno-format-nonliteral -Werror=vla -Wno-clobbered -Wdate-time -Wnull-dereference -Wfloat-conversion -std=gnu99 -MD -MT frmts/zlib/contrib/infback9/CMakeFiles/infback9.dir/infback9.c.o -MF frmts/zlib/contrib/infback9/CMakeFiles/infback9.dir/infback9.c.o.d -o frmts/zlib/contrib/infback9/CMakeFiles/infback9.dir/infback9.c.o -c /var/tmp/portage/sci-libs/gdal-3.7.0/work/gdal-3.7.0/frmts/zlib/contrib/infback9/infback9.c
FAILED: frmts/zlib/contrib/infback9/CMakeFiles/infback9.dir/infback9.c.o 
/usr/bin/x86_64-pc-linux-gnu-gcc -DDONT_DEPRECATE_SPRINTF -DGDAL_COMPILATION -DHAVE_SSE_AT_COMPILE_TIME   -mtune=generic -O2 -pipe -fvisibility=hidden -fPIC -Wall -Wextra -Winit-self -Wunused-parameter -Wmissing-prototypes -Wmissing-declarations -Wlogical-op -Wshadow -Wmissing-include-dirs -Wformat -Werror=format-security -Wno-format-nonliteral -Werror=vla -Wno-clobbered -Wdate-time -Wnull-dereference -Wfloat-conversion -std=gnu99 -MD -MT frmts/zlib/contrib/infback9/CMakeFiles/infback9.dir/infback9.c.o -MF frmts/zlib/contrib/infback9/CMakeFiles/infback9.dir/infback9.c.o.d -o frmts/zlib/contrib/infback9/CMakeFiles/infback9.dir/infback9.c.o -c /var/tmp/portage/sci-libs/gdal-3.7.0/work/gdal-3.7.0/frmts/zlib/contrib/infback9/infback9.c
In file included from /var/tmp/portage/sci-libs/gdal-3.7.0/work/gdal-3.7.0/frmts/zlib/contrib/infback9/infback9.c:6:
/var/tmp/portage/sci-libs/gdal-3.7.0/work/gdal-3.7.0/frmts/zlib/contrib/infback9/minified_zutil.h:194:45: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'OF'
  194 |     ZEXTERN uLong ZEXPORT adler32_combine64 OF((uLong, uLong, z_off_t));
      |                                             ^~
/var/tmp/portage/sci-libs/gdal-3.7.0/work/gdal-3.7.0/frmts/zlib/contrib/infback9/minified_zutil.h:195:43: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'OF'
  195 |     ZEXTERN uLong ZEXPORT crc32_combine64 OF((uLong, uLong, z_off_t));
      |                                           ^~
/var/tmp/portage/sci-libs/gdal-3.7.0/work/gdal-3.7.0/frmts/zlib/contrib/infback9/minified_zutil.h:196:47: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'OF'
  196 |     ZEXTERN uLong ZEXPORT crc32_combine_gen64 OF((z_off_t));
      |                                               ^~
`

## Operating system

Funtoo 'next'

## GDAL version and provenance

gdal-3.7.0

--- a/frmts/zlib/contrib/infback9/minified_zutil.h
+++ b/frmts/zlib/contrib/infback9/minified_zutil.h
@@ -21,6 +21,12 @@
 
 #include "zlib.h"
 
+// Gentoo and derivatives change the OF() macro from upstream zlib's zconf.h
+// to _Z_OF(), so define OF() if it isn't
+#ifndef OF
+#define OF(args) args
+#endif
+
 #if defined(STDC) && !defined(Z_SOLO)
 #  if !(defined(_WIN32_WCE) && defined(_MSC_VER))
 #    include <stddef.h>

